src/main/resources/python/docs/GenDocs.py

```diff
@@ -90,17 +92,32 @@ def run():
 					advancement = icon[index + 1:]
 					icon = icon[:index]
 
-				pages = parseBody(rawLinesIn[2:], icon, name, 2)
+				# Parse the ItemStack:entry mappings
+				mappingsRaw = rawLinesIn[2].strip().split(",")
+				mappings = []
+				for mapping in mappingsRaw:
+					if mapping == "":
+						continue
+
+					# page number within entry is optional, defaults 0
+					if ';' in mapping:
```

Comment:
As an idea:

Considering that the script is supposed to handle splitting text into pages automatically, maybe the txt format shouldn't be based on setting a page number, as the page splits are generated by the script when it runs and usually aren't known in advance when writing the txt.

Instead, could the numbers refer to forced page breaks? (The things created by an empty line or <page> tag), so that when writing the txt, you would specify 'item_id;3' and the script would understand that as 'third page break explicitly specified by the txt' and write the correct page number in the json?

As a way to make it compatible with the work you've already done, perhaps have it be 'item_id;page_number' for page number the way you already have, and 'item_id$section_number' for an easier-to-write optional new way.

Reply:
Updated based on discord discussion to instead have the numbers refer to the line which it should attempt to map the ItemStack string to, regardless of what page that line ends up on.

Updated... I'm _pretty sure_ all of the docs inputs accordingly.